tests/test_serializable_object.py

```diff
@@ -36,6 +38,71 @@ def test_cons(self):
         so.metadata['foo'] = 'bar'
         self.assertEqual(so.metadata['foo'], 'bar')
 
+    def test_cons2(self):
+        v = AnyVector()
+        v.append(1)
+        v.append('inside any vector')
+
+        d = AnyDictionary()
+        d['key_1'] = 1234
+        d['key_2'] = {'asdasdasd': 5.6}
+        so = otio.core.SerializableObjectWithMetadata(
+            metadata={
+                'string': 'myvalue',
+                'int': -999999999999,
+                'list': [1, 2.5, 'asd'],
+                'dict': {'map1': [345]},
+                'AnyVector': v,
+                'AnyDictionary': d,
+                'RationalTime': RationalTime(
+                    value=10.0,
+                    rate=5.0
+                ),
+                'TimeRange': TimeRange(
+                    RationalTime(value=1.0),
+                    RationalTime(value=100.0)
+                ),
+                'TimeTransform': TimeTransform(
```

Comment:
What about SerializableObject?

Reply:
I was starting to think that `SerializableObject` and co will cause issues, and it turns out that I was right. I'm not sure how I'll be able to make this work...

Reply:
I have the same issue with the JS bindings, which doesn't come as a surprise since I'm using the same method to convert types...

Reply:
I solved my problem. It took me almost two days of work, but we can now pass `SerializableObject`s (and its subclasses) like before.